### DISCUSSION TITLE
Why not use nested data providers?

### DISCUSSION THREAD

**Discussion by jlangree** [Budibase/budibase#10087] [Help] [answered]
From the docs:

![image](https://user-images.githubusercontent.com/48861088/226387764-0d34bec5-c061-42ca-a4b2-c14cde78f1a9.png)

What are the performance issues that could arise from nesting the data providers? For instance, would it be bad practice to nest all the data providers needed for the page at the top-levels, then nest whatever content components underneath _all_ the data providers? And if so, why?

Wouldn't it still simply be pulling each data source once regardless? I'm not challenging the point, I'm just looking for clarification.

**Comment by andz-bb** [comment, marked as answer]
Hey @jlangree 

This is describing best practice for the use of data providers. When many are being used, components within the data providers can be slow to load depending on the number of providers and the speed of the server etc. Each data provider you have on your screen will be hitting the selected data source so child components will be in their loading state while awaiting that network response. With that in mind you might find that it's best practice to only include the components that need the data provider within them. The ones that don't depend on the data provider should appear more quickly.

The general principle described there in the docs is to have more screens/modals with less complex functionality in each so you are only loading the data and components you specifically need for a given operation. I hope this helps answer what you were wanting to know! If you have any more questions on this let me know and I'll be happy to look further into it
